src/stan/callbacks/json_writer.hpp

```diff
@@ -410,13 +347,91 @@ class json_writer {
    * @param key Name of the value pair
    * @param values vector to write.
    */
-  template <typename T>
-  void write(const std::string& key, const std::vector<T>& values) {
+  void write(const std::string& key, const std::vector<std::string>& v) {
+    if (output_ == nullptr)
+      return;
+    write_sep();
+    write_key(key);
+
+    *output_ << "[ ";
+    if (v.size() > 0) {
+      auto last = v.end();
+      --last;
+      for (auto it = v.begin(); it != last; ++it) {
+        *output_ << process_string(*it) << ", ";
+      }
+    }
+    *output_ << v.back() << " ]";
+  }
+
+  /**
+   * Write a key-value pair where the value is a vector to be made a list.
+   * @param key Name of the value pair
+   * @param values vector to write.
+   */
+  void write(const std::string& key, const std::vector<double>& v) {
+    if (output_ == nullptr)
+      return;
+    write_sep();
+    write_key(key);
+
+    *output_ << "[ ";
+    if (v.size() > 0) {
+      auto last = v.end();
+      --last;
+      for (auto it = v.begin(); it != last; ++it) {
+        write_value(*it);
+        *output_ << ", ";
+      }
+      write_value(v.back());
+    }
+    *output_ << " ]";
+  }
+
+  /**
+   * Write a key-value pair where the value is a vector to be made a list.
+   * @param key Name of the value pair
+   * @param values vector to write.
+   */
+  void write(const std::string& key, const std::vector<int>& v) {
     if (output_ == nullptr)
       return;
     write_sep();
     write_key(key);
-    write_vector(values);
+
+    *output_ << "[ ";
+    if (v.size() > 0) {
+      auto last = v.end();
+      --last;
+      for (auto it = v.begin(); it != last; ++it) {
+        *output_ << *it << ", ";
+      }
+    }
+    *output_ << v.back() << " ]";
```

Comment:
Same code is replicated across multiple functions. Is there a reason you got rid of `write_vector`?

Reply:
The value type is printed differently in each case - this one uses `<<`, the vector<string> uses `process_string()`, etc. 

Write_vector was also just four functions which were the same as these but called from a template. To use this as a derived class we needed un-templated versions